Makefile

```diff
@@ -56,17 +58,20 @@ LDFLAG_OPTIONS = -ldflags "-X sigs.k8s.io/kubefed/pkg/version.version=$(GIT_VERS
 
 GO_BUILDCMD = CGO_ENABLED=0 go build $(VERBOSE_FLAG) $(LDFLAG_OPTIONS)
 
+GO_TESTCMD  = CGO_ENABLED=0 go test $(COMPILE_FLAG) $(LDFLAG_OPTIONS)
+
 TESTARGS ?= $(VERBOSE_FLAG) -timeout 60s
 TEST_PKGS ?= $(GOTARGET)/cmd/... $(GOTARGET)/pkg/...
 TEST_CMD = go test $(TESTARGS)
 TEST = $(TEST_CMD) $(TEST_PKGS)
 
 DOCKER_BUILD ?= $(DOCKER) run --rm -v $(DIR):$(BUILDMNT) -w $(BUILDMNT) $(BUILD_IMAGE) /bin/sh -c
+DOCKER_GOPATH_BUILD ?= $(DOCKER) run --rm -v $(DIR):$(GOPATH)/src/$(GOTARGET) -w $(GOPATH)/src/$(GOTARGET) $(BUILD_IMAGE) /bin/sh -c
```

Comment:
Could you help to explain more on this?  I did not see any needs to use new one than `DOCKER_BUILD`.

Reply:
@xunpan This was my plan earlier where I wanted to reuse the `DOCKER_BUILD` parameter as is. However, I ran into the following issue when I was trying to run e2e tests binary:
```
INFO: Error loading type config fixtures: Error reading fixture from path "/go/src/sigs.k8s.io/kubefed/test/common/fixtures": open /go/src/sigs.k8s.io/kubefed/test/common/fixtures: no such file or directory
```
Where my `GOPATH` is `/go`. The moment I introduced that value in `BUILDMNT` path, the issue got resolved. I would love to hear a better approach for solving this issue if you have any.

Reply:
I'm also confused - is this change required to build the binary or to run the binary that gets built?

The dependency of the test binary on fixture in the working directory for fixture appears to complicate downstream consumption of a release binary since the fixture isn't included with the release. That suggests either building fixture into the binary or reconsidering shipping the binary.